### PR TITLE
[FLINK-11013] [table] Fix distinct aggregates for group window in Table API

### PR DESCRIPTION
**Pull request by dianfu**
## What is the purpose of the change

*This pull request fix the distinct aggregate for group window on table API*

## Brief change log

*(for example:)*
  - *Update the validate logical in operators for WindowAggregate to make it work for distinct*

## Verifying this change

This change added tests and can be verified as follows:

*(example:)*
  - *Added tests in DistinctAggregateTest*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/table/DistinctAggregateTest.scala [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (walterddr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/table/DistinctAggregateTest.scala]: We can probably add these to `org.apache.flink.table.api.stream.table.AggregateTest.scala`

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/table/DistinctAggregateTest.scala [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+class DistinctAggregateTest extends TableTestBase {
+  val util = streamTestUtil()
+  val table = util.addTable[(Int, Long, String)](
+    "MyTable",
+    'a, 'b, 'c, 'proctime.proctime, 'rowtime.rowtime)
+
+  @Test
+  def testDistinctAggregate(): Unit = {
+    val result = table
```
Comment (walterddr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/stream/table/DistinctAggregateTest.scala]: this case is already covered in "AggregateTest"

**Comment by dianfu** [comment]
@walterddr Thanks a lot for your review. The comments make sense to me and have updated accordingly.

**Comment by sunjincheng121** [comment]
@dianfu @walterddr Thanks for the review and update! will merging..

**Pull request closed without merging**